part/part_test.go

```diff
@@ -594,6 +594,36 @@ func Test_prefix(t *testing.T) {
 	assert.False(t, ok)
 }
 
+func Test_deleteEmptyKey(t *testing.T) {
+	tree := New[string]()
+
+	_, _, tree = tree.Insert([]byte{}, "x")
+
+	v, watch, ok := tree.Get([]byte{})
+	assert.True(t, ok)
+	assert.Equal(t, "x", v)
+	select {
+	case <-watch:
+		t.Fatalf("channel closed")
+	default:
+	}
+
+	tree.PrintTree()
```

Comment:
do you want to keep the prints?

Reply:
Ugh I've been forgetting to remove my debug prints a lot lately. Fixed.